### PR TITLE
feat: Add a Section for Submitters

### PR DESCRIPTION
**Pull request by gmeben**
This PR:

* Adds a new "Submitter" section on the Submission Details page to show the currently assigned submitter(s)
* Improves the Jest test spec
    - Moves test data outside of assertion into its own variable
    - Adds assertions for reflecting the presence or absence of submitters

Closes Issue #555

### CODE REVIEW

File: client/src/pages/SubmissionDetails.vue [hunk 1]
```diff
@@ -4,11 +4,43 @@
...
+                  color="negative"
+                  text-color="white"
+                  icon="report_problem"
+                />
+              </q-item-section>
+              <q-item-section>
+                {{ $t("submissions.submitter.none") }}
+              </q-item-section>
+            </q-item>
+          </q-list>
```
Comment (Sruthin86) [inline client/src/pages/SubmissionDetails.vue:35]:
```suggestion
          <q-card class="my-card">
      <q-card-section>
        <p>
          <q-avatar
                  color="negative"
                  text-color="white"
                  icon="report_problem"
             />
          {{ $t("submissions.submitter.none") }}
        </p>
      </q-card-section>
    </q-card>
```

File: client/src/pages/SubmissionDetails.jest.spec.js [hunk 2]
```diff
@@ -35,60 +35,92 @@ describe("submissions details page mount", () => {
...
+
+  test("an error message appears when there are no assigned submitters", async () => {
+    await wrapper.setData({
+      submission: {
+        users: [],
+      },
+    })
+    const list = wrapper.findComponent({ ref: "list_no_submitters" })
+    expect(list.text()).toContain("submissions.submitter.none")
+    expect(list.findAllComponents({ name: "q-item" })).toHaveLength(1)
```
Comment (Sruthin86) [inline client/src/pages/SubmissionDetails.jest.spec.js:113]:
```suggestion
    expect(list.findAllComponents({ name: "q-card" })).toHaveLength(1)
```

File: client/src/pages/SubmissionDetails.jest.spec.js [hunk 3]
```diff
@@ -35,60 +35,92 @@ describe("submissions details page mount", () => {
...
 
-    expect(wrapper.findAllComponents({ name: "q-item" })).toHaveLength(3)
+  test("all assigned reviewers appear within the assigned reviewers list", async () => {
+    await wrapper.setData({
+      submission: {
+        users: submissionUsersData,
+      },
+    })
+    const list = wrapper.findComponent({ ref: "list_assigned_reviewers" })
+    expect(list.findAllComponents({ name: "q-item" })).toHaveLength(3)
```
Comment (Sruthin86) [inline client/src/pages/SubmissionDetails.jest.spec.js:123]:
```suggestion
    expect(list.findAllComponents({ name: "q-card" })).toHaveLength(3)
```

File: client/src/pages/SubmissionDetails.vue [hunk 4]
```diff
@@ -95,7 +132,12 @@
           </q-list>
         </div>
         <div v-else>
-          <q-list bordered separator data-cy="list_assigned_reviewers">
+          <q-list
```
Comment (Sruthin86) [inline client/src/pages/SubmissionDetails.vue:135]: Change the list item to a q-card

Comment (gmeben) [reply client/src/pages/SubmissionDetails.vue:135]: This is a list of reviewers, so it needs to remain as a list.

**Review by Sruthin86 — APPROVED** [review]

**Pull request merged**